### ISSUE TITLE
Build -all script

### ISSUE THREAD

**Issue by john612** [OracleSiebel/ConfiguringSiebel#43]
the text say we need to In order to build, descend into the build folder and first edit the build-all script. The scripts purpose is to build and tag containers ready to import to your local docker registry. Therefore before you start, you need to know the location of your local docker registry and you may also have a plan for naming the containers. Therefore the build script takes variables for you to specify these.  Which variables are we changing?  I am getting -bash: ./build-all: /bin/bash^M: bad interpreter: No such file or directory

**Comment by john612** [comment]
Please disregard I figured it out.

**Comment by john612** [comment]
Can you please tell me what the array values should be for the buildelement section?

**Comment by OracleSiebel** [comment]
Hi John,

Glad you figured out the first part. I'm not sure what you mean by the 'buildelement'?

**Comment by OracleSiebel** [comment]
The build-all script gives an example of how to call it:

https://github.com/OracleSiebel/ConfiguringSiebel/blob/master/Containerization/Docker/ol7/build/base_siebel/build/build-all#L2

From this you should be able to discern which values are being passed in to $1, $2 etc, and this what the parameters will be for the buildElement function.

**Comment by OracleSiebel** [comment]
Hi John,

I previously pointed you to this line:

https://github.com/OracleSiebel/ConfiguringSiebel/blob/master/Containerization/Docker/ol7/build/base_siebel/build/build-all#L2

I'll pull it here to add detail.

build-all 17.0 registry.local.com:5000 siebel registry.local.com:5000/oracle/database-instantclient/32bit:12.2.0.1

So, when you execute this line:
$1 = 17.0
$2 = registry.local.com:5000
$3 = siebel
$4 = registry.local.com:5000/oracle/database-instantclient

Thus, when the script executes, these values are used in each buildElement invocation, e.g.

buildElement cgw 17.0 registry.local.com:5000 siebel registry.local.com:5000/oracle/database-instantclient

These parameters then themselves become, for the purpose of the buildElement function:

$1 = cgw
$2 = 17.0
$3 = registry.local.com:5000
$4 = siebel
$5 = registry.local.com:5000/oracle/database-instantclient

This results in the buildElement function invoking:

docker build -f ../cgw/dockerfile -t registry.local.com/siebel/cgw:17.0 --build-arg DBCONTAINER=registry.local.com:5000/oracle/database-instantclient --squash ..

In other words, when invoke build-all you supply:

$1 = 17.0 = version you are building
$2 = registry.local.com:5000 = your local registry server address
$3 = siebel = the path in the registry server you will use for your siebel builds
$4 = registry.local.com:5000/oracle/database-instantclient = the location of your instantclient build

Does this help clarify?

**Comment by john612** [comment]
great thank you yes it helps

**Comment by john612** [comment]
I will have to document this for my team to move this into other environments and they will certainly ask about this . Thank you so much for the explanation. We have just been calling our docker files individually to create the containers.

**Comment by OracleSiebel** [comment]
OK. So I'll close this, but feel free to re-open if you need to drill into more detail on this topic.
